File: .build/plugins/outputs/skip-validator/SKIPValidator/destination/skipstone/SKIPValidator/src/main/kotlin/skipvalidator/module/SKIPValidator.kt
package skipvalidator.module

import skip.lib.*

import skip.foundation.*


class SKIPValidatorModule {




    companion object {
        /// The shared skip module.
        //    public static let shared = SKIPValidatorModule()


        fun validateEmail(email: String): Boolean {

            val emailRegEx = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,64}"
            val regex = Regex(pattern = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,64}")
            return regex.containsMatchIn(email)
        }
    }
}

File: Sources/SKIPValidator/SKIPValidator.swift
#if !SKIP
import Foundation
#else
import SkipFoundation
#endif


public struct SKIPValidatorModule {
    /// The shared skip module.
//    public static let shared = SKIPValidatorModule()
    

    public static func validateEmail(email:String) -> Bool {
        
        let emailRegEx = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,64}"
        // SKIP REPLACE:
        // val regex = Regex(pattern = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,64}")
        let emailPred = NSPredicate(format:"SELF MATCHES %@", emailRegEx)
        // SKIP REPLACE:
        // return regex.containsMatchIn(email)
        return emailPred.evaluate(with: email)
    }
    

     
}
